drivers/enc28j60/enc28j60.c

```diff
@@ -311,7 +311,7 @@ static int nd_recv(netdev_t *netdev, void *buf, size_t max_len, void *info)
 static int nd_init(netdev_t *netdev)
 {
     enc28j60_t *dev = (enc28j60_t *)netdev;
-    int res;
+    unsigned res;
```

Comment:
Res is used uninitialized on L326. I think it is missing a res = in front of the spi_init_cs function call

Reply:
True, fixed.

Reply:
Actually, spi_init_cs returns an int, and the printf on L324/L325 uses %i as well. Was the original error message about a signed vs unsigned comparison on L337?
Removing the U from the definition of STARTUP_TIMEOUT would probably fix the comparison.
Another alternative would be to introduce a new variable `unsigned count = 0` on line L334 instead. The compiler will see that `res` is unused after L325 and release the register/memory used by it. Also, the declaration of `res` could move to L323.

Reply:
Yes I agree making `STARTUP_TIMEOUT` an int would be better than continuing to create more problems.